tests/Browser/UserSearchTest.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Tests\Browser;
+
+use Tests\DuskTestCase;
+use Laravel\Dusk\Browser;
+use Tests\Browser\Pages\UserSearchPage;
+use Rogue\Models\User;
+
+class UserSearchTest extends DuskTestCase
+{
+    /**
+     * Test visiting the User Search Page.
+     * Searching for a user by name returns "No user found!" message.
+     */
+    public function testIncorrectUserSearchPage()
+    {
+        $this->browse(function (Browser $browser) {
+            $browser->loginAs(User::find(1))
+                ->visit(new UserSearchPage)
+                ->assertSeeIn('@title', 'Users');
```

Comment:
@DFurnes when I run the Dusk test just to find the title of the page matches "Users", I get this error: 
```
1) Tests\Browser\UserSearchTest::testIncorrectUserSearchPage
Facebook\WebDriver\Exception\NoSuchElementException: no such element: Unable to locate element: {"method":"css selector","selector":"body .header__title"}
  (Session info: headless chrome=59.0.3071.109)
  (Driver info: chromedriver=2.32.498513 (2c63aa53b2c658de596ed550eb5267ec5967b351),platform=Linux 4.4.0-81-generic x86_64)

/home/vagrant/DoSomething.org/rogue/vendor/facebook/webdriver/lib/Exception/WebDriverException.php:102
/home/vagrant/DoSomething.org/rogue/vendor/facebook/webdriver/lib/Remote/HttpCommandExecutor.php:320
/home/vagrant/DoSomething.org/rogue/vendor/facebook/webdriver/lib/Remote/RemoteWebDriver.php:535
/home/vagrant/DoSomething.org/rogue/vendor/facebook/webdriver/lib/Remote/RemoteWebDriver.php:175
/home/vagrant/DoSomething.org/rogue/vendor/laravel/dusk/src/ElementResolver.php:332
/home/vagrant/DoSomething.org/rogue/vendor/laravel/dusk/src/Concerns/MakesAssertions.php:270
/home/vagrant/DoSomething.org/rogue/tests/Browser/UserSearchTest.php:21
/home/vagrant/DoSomething.org/rogue/vendor/laravel/dusk/src/TestCase.php:92
/home/vagrant/DoSomething.org/rogue/tests/Browser/UserSearchTest.php:25
```

Reply:
There isn't a user with an `id` of `1` when this test is run (so it ends up trying to log in as `null` and moving on)! You can either create a user from a factory, or use the `$browser->login()` method to automatically log in as the [default user](https://github.com/DoSomething/rogue/blob/6177293cb2346313c6c49c826ed0ad0c3ac01d74/tests/DuskTestCase.php#L15-L23).

Reply:
Two handy tools for debugging Dusk tests:

1. Adding a `$browser->dump();` which will dump the HTML contents in the console (just like `$this->dump()` in an API test). 📄 
2. Adding a `$browser->screenshot('name');` which will generate a screenshot of the page in the `tests/screenshots` directory. 📸

Reply:
@DFurnes OHH I forgot it grabs from the `rogue_test` database and just grabbed from whatever database was hooked up (local/qa/prod)! Thank you!

Reply:
those are great tricks to know, thank you @DFurnes !

Reply:
All that said, it looks like switching to use `$browser->login()` creates it's own issue, since we don't have a valid Northstar token here when the search page makes calls to Northstar QA. Oy!! 

I think your best bet for now might be to actually step through the real auth flow on Northstar QA (using env vars to keep secrets out of the repo) and then we can figure out a better way to mock these external dependencies later on.

Reply:
sounds good, thanks @DFurnes !